pydundas/tests/rest/test_constant.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+from unittest.mock import patch
+from pydundas import Api
+from pydundas.rest.constant import ConstantIdError, ConstantNameError, Constant
+
+
+class TestConstant(unittest.TestCase):
+
+    # Will be used to mock the actual constants.
+    mocked_constants = {
+            'NAME': 'id',
+            'NAME1': 'iddup',
+            'NAME2': 'iddup',
+        }
+
+    def test_get_by_valid_id_returns_good_name(self):
+        with patch.dict(Constant.constants, self.mocked_constants, clear=True):
+            c = Api(None).constant()
+            self.assertEqual(c.getNamesById('id'), ['NAME'])
+
+    def test_get_by_valid_id_returns_multiple_names(self):
+        with patch.dict(Constant.constants, self.mocked_constants, clear=True):
+            c = Api(None).constant()
+            self.assertCountEqual(c.getNamesById('iddup'), ['NAME1', 'NAME2'])
+
+    def test_get_by_valid_name_returns_good_id(self):
+        with patch.dict(Constant.constants, self.mocked_constants, clear=True):
+            c = Api(None).constant()
+            self.assertEqual(c.getIdByName('name'), 'id')
+
+    def test_get_by_wrong_id_returns_ConstantIdError(self):
+        with patch.dict(Constant.constants, self.mocked_constants, clear=True):
+            c = Api(None).constant()
+            self.assertRaises(ConstantIdError, c.getNamesById, 'unknown id')
+
+    def test_get_by_wrong_name_returns_ConstantNameError(self):
+        with patch.dict(Constant.constants, self.mocked_constants, clear=True):
+            c = Api(None).constant()
+            self.assertRaises(ConstantNameError, c.getIdByName, 'unknown name')
+
+    def test_constants_become_attributes(self):
```

Comment:
As stated earlier: this basically tests the mocking framework :(
Actually, this just gave me an idea.. Since you already know that you want to test the mechanism of the Constant class where the dict is basically 'assimilated' into the class instance, why not pass it as a parameter to the init method, supplying a default value of `Constant.__retrieved_from_dundas` which would be a private dict. That way you could set up the tests using a custom dict to test whether the assimilation process functions correctly and you could also test with the actual data.

Reply:
I was ready to do that, but all rest.api* have the same pattern, and I cannot add specific parameters. One extra reason to not go meta I suppose :)